### PR TITLE
Add a full and partial display for columns in nexctl device lists

### PR DESCRIPTION
**Pull request by nerdalert**
issue #1112

### CODE REVIEW

**Comment by nerdalert** [comment]
Output examples for the four different device list options:

```

# List devices

./nexctl-dev-linux-arm64 --host https://api.try.nexodus.127.0.0.1.nip.io --username admin --password floofykittens device list
DEVICE ID                                HOSTNAME                     NODE ADDRESS IPV4     NODE ADDRESS IPV6     ENDPOINT IP            REFLEXIVE IPv4          ORGANIZATION ID                          OS         RELAY
8e5f75fb-9866-44c6-879b-23f808058985     node-2                       100.100.0.2           200::2                192.168.1.9:60477      [x.x.x.33:60477]        2c493b68-84cf-4bf6-8e1e-4412497fc890     darwin     false
36dc9771-f7a9-41e4-82c3-57ebc5363424     node-1                       100.100.0.1           200::1                192.168.64.2:46249     [x.x.x.33:40194]        2c493b68-84cf-4bf6-8e1e-4412497fc890     linux      false

# List devices wide
./nexctl-dev-linux-arm64 --host https://api.try.nexodus.127.0.0.1.nip.io --username admin --password floofykittens device list -f
DEVICE ID                                HOSTNAME                     NODE ADDRESS     ENDPOINT IP            PUBLIC KEY                                       ORGANIZATION ID                          LOCAL IP               ALLOWED IPS                     TUNNEL IPV4     TUNNEL IPV6     CHILD PREFIX     ORG PREFIX IPV4     ORG PREFIX IPV6     REFLEXIVE IPv4          ENDPOINT LOCAL IPv4     OS         SECURITY GROUP ID                        RELAY
8e5f75fb-9866-44c6-879b-23f808058985     node-2                       100.100.0.2      192.168.1.9:60477      xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx       2c493b68-84cf-4bf6-8e1e-4412497fc890     192.168.1.9:60477      [100.100.0.2/32 200::2/128]     100.100.0.2     200::2          []               100.100.0.0/16      200::/64            [x.x.x.33:60477]        192.168.1.9             darwin     39bba205-be91-4a60-94e0-b55d399ddcdb     false
36dc9771-f7a9-41e4-82c3-57ebc5363424     node-1                       100.100.0.1      192.168.64.2:46249     yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy       2c493b68-84cf-4bf6-8e1e-4412497fc890     192.168.64.2:46249     [100.100.0.1/32 200::1/128]     100.100.0.1     200::1          []               100.100.0.0/16      200::/64            [x.x.x.33:40194]        192.168.64.2            linux      39bba205-be91-4a60-94e0-b55d399ddcdb     false


# List devices in org

./nexctl-dev-linux-arm64 --host https://api.try.nexodus.127.0.0.1.nip.io --username admin --password floofykittens device list --organization-id 2c493b68-84cf-4bf6-8e1e-4412497fc890
DEVICE ID                                HOSTNAME                     NODE ADDRESS IPV4     NODE ADDRESS IPV6     ENDPOINT IP            REFLEXIVE IPv4          ORGANIZATION ID                          OS         RELAY
8e5f75fb-9866-44c6-879b-23f808058985     node-2                       100.100.0.2           200::2                192.168.1.9:60477      [x.x.x.33:60477]        2c493b68-84cf-4bf6-8e1e-4412497fc890     darwin     false
36dc9771-f7a9-41e4-82c3-57ebc5363424     node-1                       100.100.0.1           200::1                192.168.64.2:46249     [x.x.x.33:40194]        2c493b68-84cf-4bf6-8e1e-4412497fc890     linux      false

# List devices in org wide
./nexctl-dev-linux-arm64 --host https://api.try.nexodus.127.0.0.1.nip.io --username admin --password floofykittens device list --organization-id 2c493b68-84cf-4bf6-8e1e-4412497fc890  -f
DEVICE ID                                HOSTNAME                     NODE ADDRESS     ENDPOINT IP            PUBLIC KEY                                       ORGANIZATION ID                          LOCAL IP               ALLOWED IPS                     TUNNEL IPV4     TUNNEL IPV6     CHILD PREFIX     ORG PREFIX IPV4     ORG PREFIX IPV6     REFLEXIVE IPv4          ENDPOINT LOCAL IPv4     OS         SECURITY GROUP ID                        RELAY
8e5f75fb-9866-44c6-879b-23f808058985     node-2                       100.100.0.2      192.168.1.9:60477      xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx       2c493b68-84cf-4bf6-8e1e-4412497fc890     192.168.1.9:60477      [100.100.0.2/32 200::2/128]     100.100.0.2     200::2          []               100.100.0.0/16      200::/64            [x.x.x.33:60477]        192.168.1.9             darwin     39bba205-be91-4a60-94e0-b55d399ddcdb     false
36dc9771-f7a9-41e4-82c3-57ebc5363424     node-1                       100.100.0.1      192.168.64.2:46249     yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy       2c493b68-84cf-4bf6-8e1e-4412497fc890     192.168.64.2:46249     [100.100.0.1/32 200::1/128]     100.100.0.1     200::1          []               100.100.0.0/16      200::/64            [x.x.x.33:40194]        192.168.64.2            linux      39bba205-be91-4a60-94e0-b55d399ddcdb     false

```

**Comment by russellb** [comment]
This is a preference thing, but I still find the default output too wide. How about:

```
DEVICE ID       HOSTNAME       NODE ADDRESSES         RELAY         ORG ID
```

**Comment by nerdalert** [comment]
> This is a preference thing, but I still find the default output too wide. How about:
> 
> ```
> DEVICE ID       HOSTNAME       NODE ADDRESSES         RELAY         ORG ID
> ```

wfm, also changed some naming to match between both option on the amend.

File: cmd/nexctl/device.go [hunk 1]
```diff
@@ -9,26 +9,47 @@ import (
...
 				if endpoint.Source == "local" {
 					localIp = endpoint.Address
-					break
+				} else {
+					reflexiveIp4 = append(reflexiveIp4, endpoint.Address)
 				}
 			}
-			fmt.Fprintf(w, fs, dev.Id, dev.Hostname, dev.TunnelIp, dev.TunnelIpV6, localIp, dev.PublicKey, dev.OrganizationId, dev.Os, fmt.Sprintf("%t", dev.Relay))
+			if !fullDisplay {
+				fmt.Fprintf(w, fs, dev.Id, dev.Hostname, dev.TunnelIp, dev.OrganizationId, fmt.Sprintf("%t", dev.Relay))
```
Comment (russellb) [inline cmd/nexctl/device.go:47]:
```suggestion
				fmt.Fprintf(w, fs, dev.Id, dev.Hostname, fmt.Sprintf("[%s %s]", dev.TunnelIp, dev.TunnelIpV6), dev.OrganizationId, fmt.Sprintf("%t", dev.Relay))
```

File: cmd/nexctl/device.go [hunk 2]
```diff
@@ -43,35 +64,46 @@ func listOrgDevices(c *public.APIClient, organizationID uuid.UUID, encodeOut str
...
-		if encodeOut != encodeNoHeader {
-			fmt.Fprintf(w, fs, "DEVICE ID", "HOSTNAME", "NODE ADDRESS",
+		var fs string
+		if fullDisplay {
+			fs = "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n"
+		} else {
+			fs = "%s\t%s\t%s\t%s\t%s\n"
+		}
+		if encodeOut != encodeNoHeader && !fullDisplay {
+			fmt.Fprintf(w, fs, "DEVICE ID", "HOSTNAME", "TUNNEL IPV4", "ORGANIZATION ID", "RELAY")
```
Comment (russellb) [inline cmd/nexctl/device.go:81]:
```suggestion
			fmt.Fprintf(w, fs, "DEVICE ID", "HOSTNAME", "TUNNEL IPS", "ORGANIZATION ID", "RELAY")
```

**Review by russellb — APPROVED** [review]

**Pull request merged**